numpy/_core/src/multiarray/ctors.c

```diff
@@ -2114,7 +2114,7 @@ _is_default_descr(PyObject *descr, PyObject *typestr) {
     if (!PyList_Check(descr) || PyList_GET_SIZE(descr) != 1) {
         return 0;
     }
-    PyObject *tuple = PyList_GET_ITEM(descr, 0);
+    PyObject *tuple = PyList_GET_ITEM(descr, 0); // noqa: borrowed-ref - manual fix needed
```

Comment:
Ok so this is using an external `PyObject`, which made me think this might need a manual fix. I think this is ok as is, but I'm unsure.